Sources/Basic/OutputByteStream.swift

```diff
@@ -267,84 +235,94 @@ precedencegroup StreamingPrecedence {
+    (value as ByteStreamable).write(to: stream)
     return stream
 }
 
-@discardableResult
-public func <<< <C: Collection>(stream: OutputByteStream, value: C)
-    -> OutputByteStream where C.Element == UInt8 {
-    stream.write(collection: value)
-    return stream
+extension UInt8: ByteStreamable {
+    public func write(to stream: OutputByteStream) {
+        stream.write(self)
+    }
 }
 
-@discardableResult
-public func <<< <S: Sequence>(
-    stream: OutputByteStream,
-    value: S
-) -> OutputByteStream where S.Iterator.Element == UInt8 {
-    stream.write(sequence: value)
-    return stream
+extension Character: ByteStreamable {
+    public func write(to stream: OutputByteStream) {
+        stream.write(String(self))
+    }
 }
 
-@discardableResult
-public func <<< (stream: OutputByteStream, value: String) -> OutputByteStream {
-    stream.write(value)
-    return stream
+extension String: ByteStreamable {
+    public func write(to stream: OutputByteStream) {
+        stream.write(self.utf8)
+    }
 }
 
-@discardableResult
-public func <<< (stream: OutputByteStream, value: Character) -> OutputByteStream {
-    stream.write(value)
-    return stream
+extension Substring: ByteStreamable {
+    public func write(to stream: OutputByteStream) {
+        stream.write(self.utf8)
+    }
 }
 
-@discardableResult
-public func <<< (stream: OutputByteStream, value: ByteStreamable) -> OutputByteStream {
-    stream.write(value)
-    return stream
+extension StaticString: ByteStreamable {
+    public func write(to stream: OutputByteStream) {
+        withUTF8Buffer { stream.write($0) }
+    }
 }
 
-@discardableResult
-public func <<< (stream: OutputByteStream, value: TextOutputStreamable) -> OutputByteStream {
-    stream.write(value)
-    return stream
+extension Array: ByteStreamable where Element == UInt8 {
+    public func write(to stream: OutputByteStream) {
+        stream.write(self)
+    }
 }
 
-extension UInt8: ByteStreamable {
+extension ArraySlice: ByteStreamable where Element == UInt8 {
     public func write(to stream: OutputByteStream) {
         stream.write(self)
     }
 }
 
-extension Character: ByteStreamable {
+extension ContiguousArray: ByteStreamable where Element == UInt8 {
     public func write(to stream: OutputByteStream) {
         stream.write(self)
     }
 }
 
-extension String: ByteStreamable {
+extension UnsafeBufferPointer: ByteStreamable where Element == UInt8 {
+    public func write(to stream: OutputByteStream) {
+        stream.write(self)
+    }
+}
+
+extension UnsafeMutableBufferPointer: ByteStreamable where Element == UInt8 {
+    public func write(to stream: OutputByteStream) {
+        stream.write(self)
+    }
+}
+
+extension UnsafeRawBufferPointer: ByteStreamable {
+    public func write(to stream: OutputByteStream) {
+        stream.write(self)
+    }
+}
+
+extension UnsafeMutableRawBufferPointer: ByteStreamable {
```

Comment:
Are we using the pointer types somewhere?

Reply:
Nope: I was just thinking that in the spirit of Basic/Utility becoming separate libraries it would make sense to provide `ByteStreamable` conformances for common Standard Library types. Similar to the `Sequence` overload: we don't use it anywhere.

Reply:
I am a little nervous with adding these right now. We can always expand when we need.

Reply:
I've removed them in the latest PR.